freqtrade/freqtradebot.py

```diff
@@ -1140,33 +1141,78 @@ def check_handle_timedout(self) -> None:
+        :param trade: Trade object.
+        :return: None
+        """
+        if order['side'] == trade.entry_side:
+            self.handle_cancel_enter(trade, order, constants.CANCEL_REASON['TIMEOUT'])
+        else:
+            canceled = self.handle_cancel_exit(
+                trade, order, constants.CANCEL_REASON['TIMEOUT'])
+            canceled_count = trade.get_exit_order_count()
+            max_timeouts = self.config.get('unfilledtimeout', {}).get('exit_timeout_count', 0)
+            if canceled and max_timeouts > 0 and canceled_count >= max_timeouts:
+                logger.warning(f'Emergency exiting trade {trade}, as the exit order '
+                               f'timed out {max_timeouts} times.')
+                try:
+                    self.execute_trade_exit(
+                        trade, order['price'],
+                        exit_check=ExitCheckTuple(exit_type=ExitType.EMERGENCY_EXIT))
+                except DependencyException as exception:
+                    logger.warning(
+                        f'Unable to emergency sell trade {trade.pair}: {exception}')
+
+    def replace_orders(self, order: Dict, order_obj: Optional[Order], trade: Trade) -> None:
+        """
+        Check if any orders should be replaced and do so
+        :param order: Order dict grabbed with exchange.fetch_order()
+        :param order_obj: Order object.
+        :param trade: Trade object.
+        :return: None
+        """
+        analyzed_df, _ = self.dataprovider.get_analyzed_dataframe(trade.pair,
+                                                                  self.strategy.timeframe)
+        latest_candle_open_date = analyzed_df.iloc[-1]['date'] if len(analyzed_df) > 0 else None
+        latest_candle_close_date = timeframe_to_next_date(self.strategy.timeframe,
+                                                          latest_candle_open_date)
+        # Check if new candle
+        if order_obj and latest_candle_close_date.replace(tzinfo=None) > order_obj.order_date:
+            # New candle
+            proposed_rate = self.exchange.get_rate(
+                trade.pair, side='entry', is_short=trade.is_short, refresh=True)
+            adjusted_entry_price = strategy_safe_wrapper(self.strategy.readjust_entry_price,
+                                                         default_retval=proposed_rate)(
+                pair=trade.pair, current_time=datetime.now(timezone.utc),
+                proposed_rate=proposed_rate, entry_tag=trade.enter_tag,
+                side=trade.entry_side)
+            # check if user has requested entry limit adjustment
+            if proposed_rate != adjusted_entry_price:
+                # cancel existing order
+                self.handle_cancel_enter(trade, order, constants.CANCEL_REASON['REPLACE'],
+                                         allow_full_cancel=False)
+                stake = self.wallets.get_trade_stake_amount(trade.pair, self.edge)
```

Comment:
i actually don't think this is correct. The code will work - but usage of this is most likely wrong.

It'll work "most of the time" assuming no concurrency and no partially filled orders - but:

* if an additional stake became available - then the "new" entry might have a higher stake than what you had initially (which is potentially unwanted). It'll depend on the configuration - but is tricky to fully test or even think through, as you can have quite some different scenarios (all setting combinations + actual trade scenarios).
* if the order filled partially (say 80/100) - then you're now creating a new order for another 100 units - so your final position is 180 - instead of the 100 you aimed for.

I think the better approach might be to capture the "full" value of the initial order (say you were going for 100 - and filled 80) - then you'd try to place a new order for 20 - so in the end, your "full" stake is filled

Reply:
I see your point. For this reason I'm considering changing the return of the new strategy callback to have: `float price` + `bool confirmation`. Adding this extra return will remove the need of comparing proposed stake vs new user suggested stake as a means of detecting user intention, thus removing the need of queriying `get_trade_stake_amount`. Added benefit is that it would also allow user to more easily indicate a sort of `timeout` for these readjusted orders.
Looking forward for your feedback on this.

Reply:
Seems I've also confused stake and rate in my previous comment. I agree, replacement order should use the previous attempted(canceled) stake ammout.

Reply:
Afterthough on `bool confirmation`. Probably not needed, we could use `None` to indicate that we don't want order adjustment.

Reply:
i'm not sure this is final yet (see other comment) ...